src/Sylius/Bundle/ApiBundle/spec/Validator/CatalogPromotion/ForProductsScopeValidatorSpec.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\Sylius\Bundle\ApiBundle\Validator\CatalogPromotion;
+
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use Sylius\Bundle\ApiBundle\SectionResolver\AdminApiSection;
+use Sylius\Bundle\ApiBundle\SectionResolver\ShopApiSection;
+use Sylius\Bundle\CoreBundle\SectionResolver\SectionProviderInterface;
+use Sylius\Bundle\CoreBundle\Validator\CatalogPromotionScope\ScopeValidatorInterface;
+use Sylius\Bundle\CoreBundle\Validator\Constraints\CatalogPromotionScope;
+use Sylius\Component\Core\Model\ProductInterface;
+use Sylius\Component\Core\Repository\ProductRepositoryInterface;
+use Symfony\Component\Validator\Context\ExecutionContextInterface;
+use Symfony\Component\Validator\Violation\ConstraintViolationBuilderInterface;
+
+final class ForProductsScopeValidatorSpec extends ObjectBehavior
+{
+    function let(ScopeValidatorInterface $baseScopeValidator, SectionProviderInterface $sectionProvider): void
+    {
+        $this->beConstructedWith($baseScopeValidator, $sectionProvider);
+    }
+
+    function it_is_a_scope_validator(): void
+    {
+        $this->shouldHaveType(ScopeValidatorInterface::class);
+    }
+
+    function it_fallbacks_to_default_validator_if_it_is_not_api_section(
+        ScopeValidatorInterface $baseScopeValidator,
+        SectionProviderInterface $sectionProvider,
+        ExecutionContextInterface $executionContext,
+    ): void {
+        $constraint = new CatalogPromotionScope();
+
+        $sectionProvider->getSection()->willReturn(new ShopApiSection());
+
+        $baseScopeValidator->validate([], $constraint, $executionContext)->shouldBeCalled();
+
+        $executionContext->buildViolation(Argument::any())->shouldNotBeCalled();
+
+        $this->validate([], $constraint, $executionContext);
+    }
+
+    function it_adds_violation_if_catalog_promotion_scope_does_not_have_products_key_configured(
```

Comment:
I'm not sure it is worth testing, but we have two cases for all these checks: 
* not existence (covered)
* empty value (not covered)
Which could be added, as variant case is covered fully. Same is true for taxon